src/battle-tooltips.ts

```diff
@@ -327,23 +332,26 @@ class BattleTooltips {
 			}
 			let serverPokemon = this.battle.myPokemon![activeIndex];
 			buf = this.showPokemonTooltip(pokemon, serverPokemon);
+			tooltipType = 'switch-pokemon';
 			break;
 		}
 		case 'field': {
 			buf = this.showFieldTooltip();
+			tooltipType = 'field';
 			break;
 		}
 		default:
 			// "throws" an error without crashing
 			Promise.resolve(new Error(`unrecognized type`));
 			buf = `<p class="message-error" style="white-space: pre-wrap">${new Error(`unrecognized type`).stack}</p>`;
+			tooltipType = 'unknown';
 		}
 
-		this.placeTooltip(buf, elem, ownHeight);
+		this.placeTooltip(buf, elem, ownHeight, tooltipType);
```

Comment:
I wonder if you'd prefer to just put the `type` itself here, instead of making a separate `tooltipType` variable?

Reply:
I don't really mind, as long as there's classes in them, I'm good. Though you'd have to take it from here since my JS knowledge is limited.